lib/sinon/collection.js

```diff
@@ -46,7 +46,16 @@ var collection = {
     },
 
     resetHistory: function resetHistory() {
-        each(this, "resetHistory");
+        getFakes(this).forEach(function (fake) {
+            if (typeof fake.resetHistory === "function") {
+                fake.resetHistory();
+                return;
```

Comment:
OK, I'll take the bait.

If follow your original suggestion, while writing JavaScript without omitting syntax, we'll end up with this:

```javascript
getFakes(this).forEach(function (fake) {
    if (typeof fake.resetHistory === "function") {
        fake.resetHistory();
    } else {
        if (typeof fake.reset === "function") {
            fake.reset();
        }
    }
});
```

In my opinion, that requires a lot more effort to understand.

Using [`return`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/return) as a control flow statement helps me keep indentation low, which I think we all agree makes code easier to read. 

Regarding consistency: I don't think we're that consistent in this codebase. We've been using `return` as a control flow statement across this codebase for awhile, try this search:

```shell
$ ag --ignore=docs/ -Q "return;"
```

I am all for consistency across a codebase ... until we find a better way of doing things, then we should slowly refactor to adopt the better practice. I believe that using `return` to reduce indentation is a good practice.

> How would you feel adding a (pointless) return statement to the other if for consistency's sake?

ESLint understands how to keep [consistent return](http://eslint.org/docs/rules/consistent-return), which [is configured in `eslint-config-sinon`](https://github.com/sinonjs/eslint-config-sinon/blob/master/index.js#L118).

I think it is known to any (non-novice) reader, that the default return value of a function `undefined`.

\</bike-shed\>